Example/Example 1/sampleserver.js

```diff
@@ -89,13 +91,30 @@ io.on('connection', (client) => {
 
     // When a player closes the game or refresh the page, this event will be triggered
     client.on('disconnect', () => {
+        try {
+            // Tell everyone that we disconnected (ourself NOT included, because we already closed the game and we don't care)
+            client.broadcast.emit('destroy_player', player.toString());
 
-        // Tell everyone that we disconnected (ourself NOT included, because we already closed the game and we don't care)
-        client.broadcast.emit('destroy_player', player.toString());
+            //Remove player from list
+            players.splice(players.indexOf(player), 1);
 
-        //Remove player from list
-        players.splice(players.indexOf(player), 1);
+            console.log(`Player "${player.username}", with ID: ${player.id} disconnected.`);
+        }
+        catch (err) {
+            //If the player is undefined we need to remove him here
+            try {
+                let ids = [];
+                io.sockets.sockets.forEach(socket => {
+                    ids.push(socket.id);
+                });
+                players = players.filter(player => {
+                    return ids.includes(player.socketId);
+                });
+            }
+            catch (error) {
+                console.log(error)
+            }
+        }
```

Comment:
Thats what I first assumed as well, but it also seems to happen in other circumstances where the player played for some time and therefore also is in the player's array, so if we just return the function we would have somewhat of a memory leak since he would never get removed from the Array.
A solution that would work is to check if is undefined and then do the trick which I had in the catch block that checks if any player in the players array isn't connected.

Reply:
Its hard to say since it happens so irregularly but the way i found it out was that my game mobabattle.io which has a few thousand players a day. Had a problem where there were "ghost" in lobbies meaning people who were still in the players array but not connected to the server. I just had a try and catch block arround my on.disconnect function which basically does the same as the (!player). function. And "ghost" can only be created if the once were players but the disconnect function fails to write them out so there has to be a scenario where we need my addition. But i really dont know what causes the exact problem.

Reply:
Using my solution removed the problem in its entirely, but there could be a better way.

Reply:
weird, maybe socketio does not trigger the disconnect event at some edge cases. Are you using the latest socketio version? There is a v4 out already, this extension is still running on v3 --> #13 

i remember that i encountered a case where a callback was not called due to missing arguments in my callback function, apparently the disconnect event expects a `reason` argument https://socket.io/docs/v3/server-api/#event-disconnect

Reply:
but to really hunt down the bug I would need a reproduceable project, there can be many reasons for that, especially in a bigger project like yours. The fix is curing the symptoms and not actually root cause :(

Reply:
I'm using the newest version, but I'm pretty sure it also was a problem with the v3 version.
But the weird thing is if it would be because of a reason argument missing. Wouldn't it just not go into the disconnect event?
But it does go into it disconnect event every time, but the player is undefined in some cases.
I really don't know, since I don't have that much knowledge about how socket.io truly works.
My cleanup commits fixes the issues and is pretty clean maybe just merge this one for now look deeper into it if you feel like it ^^.

Reply:
> but to really hunt down the bug I would need a reproduceable project, there can be many reasons for that, especially in a bigger project like yours. The fix is curing the symptoms and not actually the root cause :(

Yeah I agree not that satisfying

Reply:
> But the weird thing is if it would be because of a reason argument missing. Wouldn't it just not go into the disconnect event?

idk socket.io well enough under the hood as well, but if there are method overloads for example it wont match your callback signature and therefor might not call it in some circumstances. I'm just thinking out loud, if the player becomes undefined over some amount of time after being initialized before, than you might be setting the player to undefined somewhere in your code.

Anyway, nice that you found a solution for your case and thanks for the PR! I will apply the changes except for the workaround as it feels hacky to me and I dont think this should be in the example server. If i find some time I will provide a new example server in typescript as I also have a .io project in the works